src/reconstruction/refinement/updateMetNames.m

```diff
@@ -0,0 +1,30 @@
+function [updatedModel] = updateMetNames(originalModel, modelToUpdate)
+% Updates model.metNames in a new model, using metNames of an original
+% model.
+%
+% USAGE:
+%
+%    [updatedModel] = updateMetNames(originalModel, modelToUpdate)
+%
+% INPUTS:
+%    originalModel:      COBRA model structure with correct model.metNames
+%    modelToUpdate:      COBRA model structure that needs to have its
+%                        model.metNames updated 
+%
+% OUTPUT:
+%    updatedModel:       COBRA model structure with corrected metNames
+%
+% .. Authors:
+%       - written by Diana El Assal 27/06/2017
+
+indMets = findMetIDs(originalModel, modelToUpdate.mets);
```

Comment:
This could (potentially) be done indenpendent of the compartment (i.e. creating a metID -> metName matching) which could cover a larger fraction if the new model has additional compartments.